### PR TITLE
feat(ci): ensure stable devshells in pc node

### PR DESCRIPTION
**Pull request by tgunnoe**
This adds a step to build partner-chains outputs with the current PR on smart-contracts in order to always ensure stable devshells on partner-chains.

### Prereview checklist

- [ ] The [CHANGELOG.md](../blob/master/CHANGELOG.md) has been updated under the `# Unreleased` header using the appropriate sub-headings with links to the appropriate issues/PRs.
- [x] All tests pass in CI
- [ ] PR was self-reviewed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,79 +30,67 @@ jobs:
...
-    permissions:
-      id-token: write
-      contents: read
-    runs-on: [ self-hosted, macOS ]
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-        with:
-          token: ${{ github.token }}
+          repository: input-output-hk/partner-chains
```
Comment (nrdxp) [inline .github/workflows/ci.yml:61]: does this work as is? The checkout action uses the autogenerated github token by default, which only has access to the current repo (not any other repos in the org). You might need to use a different token here, one with access to the pc repo.

Comment (tgunnoe) [reply .github/workflows/ci.yml:61]: It worked on a previous run. does it need anything if the repo is public? I don't think so

**Comment by gilligan** [comment]
I'm confused about what this is or what it aims to achieve? why are we checking out `partner-chains` ? I don't believe we should create a dependency in this direction

**Comment by tgunnoe** [comment]
> I'm confused about what this is or what it aims to achieve? why are we checking out partner-chains ? I don't believe we should create a dependency in this direction

It's to prevent the scenarios last week when a merge here broke the devshells on partner-chains. To ensure changes wont break downstream shells

**Review by gilligan — CHANGES_REQUESTED** [review]
This basically introduces some sort of circular dependency between pcsc and pc - I really don't think this is the way we should be going at all.
We can have some local check to ensure that the devShell from pcsc has the mandatory tools in $PATH but nothing beyond that.
See https://github.com/input-output-hk/partner-chains-smart-contracts/blob/pc-redo-nix/nix/tests/test-shell.sh for a possible approach

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,79 +30,67 @@ jobs:
...
-          nix run github:srid/nixci -- -v build -- --fallback > ${{ runner.temp }}/outputs
-
-      - name: Acquire AWS credentials
-        uses: aws-actions/configure-aws-credentials@v4
-        with:
-          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
-          aws-region: ${{ env.AWS_DEFAULT_REGION }}
-
-      - name: Copy nix scopes to nix cache
+          nix flake lock --update-input trustless-sidechain \
```
Comment (LGLO) [inline .github/workflows/ci.yml:66]: In partner-chains repo I've renamed trustless-sidechain to smart-contracts. Hence if this PR has been merged, I would have broken the main workflow of a _dependency_.

**Pull request closed without merging**